### PR TITLE
Our first bit of code, introducing react and django 

### LINKED ISSUE #1: Set up readme
We have been floating around a lot of ideas about what APIs, frameworks, and services to use.  We should settle on which ones to use and detail how to set them up on a local machine and how to program with them, or at least links to their documentation on how to do these things.  This is probably an issue for two people, each focusing on a few aspects of setup that they are familiar with (like Jake with django, or whoever else with the Spotify API, preferably someone with a spotify account, if we are using that).  At the very least, we need a project summary, some basic goals, everyone's name, and links to APIs and services we will most likely be using.

### LINKED ISSUE #6: Initialize code base with Django/React
Commit the starter code for Django/React, this includes files like app.js, index.html (for react). This can be done with npx create-react-app musicGames in the terminal. For Django, this can be done with django-admin startproject musicGames. I'm not sure if we've officially decided on a framework,m since the google doc lists both Django and React.

### PR DESCRIPTION
**Pull request by A-Waters**
Hello, after a few hours of work here is what we got. This should deal with issue #6 

The project is broken down into 2 main sections
The front end and the back end.

The front end is React. Specifically in the GUI folder is where you will find your basic react files. 
The back end is where we can find our python environment (env) and Django DB files (src).

The front side and backside should be able to run independently (I believe). The way the front end and back end interact is by using a package called "axios" and using GET methods to get access to the data. (e.g. the frontend/gui/src/components/list.js componentDidMount() method)

people will also have to install node.js (https://nodejs.org/en/) (LTS Perferably) (this will take a few minuets) if they are working with react.

Make sure to install the following packages.
pip is for the backside of the website, 
npm is for the frontside of the website
```
pip install django djangorestframework virtualenv pep8 autopep8 pylint django-cors-headers
npm install axios
```

you should be able to start up the backside by running the command:
```
python backend\src\manage.py runserver
```
you should be able to run the frontside by running the commands:
```
cd .\frontend\gui
npm install
npm start
```

if you have any questions or errors starting up lmk.
Someone should also update the read me with all this information. I put some in allready just to keep track of it but its not done nicely.

### CODE REVIEW

**Comment by A-Waters** [comment]
I just realized that the front side is missing node modules, working on workaround so we don't have to deal with it. LMK if you have any ideas. (resolved)

**Comment by JakeCapra** [comment]
With react, you have to locally install the modules.
`npm install`
I had the same issue on a previous assignment.

**Comment by A-Waters** [comment]
Thank you (updated initial comment)

**Comment by lenora4321** [comment]
Is it just me, or is there no package.json file, which npm needs to start?

**Comment by jbuzzell** [comment]
@lenora4321 the package.json file is included in the npm install, you have to be in the directory you're running from when you run npm install axios

**Comment by lenora4321** [comment]
For some reason it's in my frontend folder. Welp
So where should I be installing this all from?

**Comment by jbuzzell** [comment]
@lenora4321 frontend/gui

**Comment by lenora4321** [comment]
Oh yeah you're right nevermind.  Thanks

**Comment by A-Waters** [comment]
Updated initial comment for more clarification.
also when you guys give the thumbs up we will merge into master. I'm not an expert on Django or react. @lenora4321, as far as I know, you're the most experienced in this group with react(frontend/gui/) and @jbuzzell you probably (almost definitely) know the most about Django DB (backend/src/) so as long as you each know what's happening in those directories I think we are good to merge.

**Comment by lenora4321** [comment]
Haha that's funny because I've never used react in my life.

**Comment by A-Waters** [comment]
Well then we're in new territory MB, Its all g. we should just figure out who really knows react then. Im iffy on it but I think its okay rn.

**Comment by lenora4321** [comment]
Alright, so I've made it to the screen that says "check console".  Is there something else I'm supposed to be checking in this code, because it works up to there.  What's supposed to be in the console?

**Comment by A-Waters** [comment]
Nah if you got that far it looks good. In the console, if you have the backside running and you added some an article to the DB then you would see it there. But the front side is working and rn that's all that matters.

**Comment by JakeCapra** [comment]
> Updated initial comment for more clarification.
> also when you guys give the thumbs up we will merge into master. I'm not an expert on Django or react. @lenora4321, as far as I know, you're the most experienced in this group with react(frontend/gui/) and @jbuzzell you probably (almost definitely) know the most about Django DB (backend/src/) so as long as you each know what's happening in those directories I think we are good to merge.

Nate and I used React in the last project.

**Comment by A-Waters** [comment]
Yeah, same but that is about my only experience with it. We can work it out I'm not worried.

**Review by lenora4321 — APPROVED** [review]
Seems good to me, as long as we get clarification on what directories to run what commands from in the future, which I think is already being done in another issue.  Good work!

**Comment by JakeCapra** [comment]
Yeah, we need to be more specific with the directions. Other than that it's good, nice work.

**Review by JakeCapra — APPROVED** [review]

File: backend/src/articles/api/views.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from rest_framework.generics import ListAPIView, RetrieveAPIView
+
+from articles.models import Article
+from .serializers import ArticalSerialziers
+
+
+class AricalsListView(ListAPIView):
```
Comment (jbuzzell) [inline backend/src/articles/api/views.py:7]: watch out for misspellings in code or have someone proofread if it's something you have difficulties with (i.e. dyslexia), if we catch them early, the less we have to refactor.

**Review by jbuzzell — APPROVED** [review]
ok, finally got everything set up. i ended up deleting and re-cloning the repo. no idea how that fixed anything, but here we are. 

django seems to be doing just fine. i hadn't fully grasped how it was being set up--i have never used the REST API before for a project. django is usually standalone & full-stack; this use case of using django's MANY database tools and just grabbing the data from it on a different port is both really interesting and actually perfect for what we're doing--we don't have to fuck around with setting up SQL/sqlite, django does it for us. 

looks like it was set up fine. for the future, as i recall you mentioning this: all database admin is done through django's admin site (/admin). you can create an administrator and create users and database objects in that menu. https://docs.djangoproject.com/en/3.0/intro/tutorial02/#introducing-the-django-admin

File: backend/src/articles/api/views.py (hunk 1 above)
Comment (A-Waters) [reply backend/src/articles/api/views.py:7]: will look out for them, ill proabbyl ask somoeone for the help, spelling really isnt my best field of work tho.

**Pull request merged**